add SL rn (#4668)
fishtown-analytics/docs.getdbt.com: website/docs/docs/dbt-versions/release-notes/74-Dec-2023/dec-sl-updates.md

---
title: "Updates and fixes: dbt Semantic Layer and MetricFlow updates for December 2023."
description: "December 2023: Enhanced Tableau integration, BIGINT support, LookML to MetricFlow conversion, and deprecation of legacy features."
sidebar_label: "Update ad fixes: dbt Semantic Layer and MetricFlow."
sidebar_position: 08
date: 2023-12-22
---
The dbt Labs team continues to work on adding new features, fixing bugs, and increasing reliability for the dbt Semantic Layer and MetricFlow. Here are the updates and fixes for December 2023.

## Bug fixes
- The dbt Semantic Layer integration with Tableau now supports queries that resolve to a "NOT IN" clause (for example: using "exclude" in the filtering user interface). Previously it wasn’t supported.

> - The dbt Semantic Layer can support `BIGINT` with over 18 digits. Previously it would return an error.

Reviewer: I'd say "precision larger than 18" instead of "over 18 digits".

---

> title: "dbt Semantic Layer and MetricFlow updates for December 2023"

Author: ```suggestion
title: "dbt Semantic Layer updates for December 2023"
```

description: "December 2023: Enhanced Tableau integration, BIGINT support, LookML to MetricFlow conversion, and deprecation of legacy features."

> sidebar_label: "Update and fixes: dbt Semantic Layer and MetricFlow"

Author: ```suggestion
sidebar_label: "Update and fixes: dbt Semantic Layer"
```

sidebar_position: 08
date: 2023-12-22
---

> The dbt Labs team continues to work on adding new features, fixing bugs, and increasing reliability for the dbt Semantic Layer and MetricFlow.

Author: ```suggestion
The dbt Labs team continues to work on adding new features, fixing bugs, and increasing reliability for the dbt Semantic Layer.
```

Refer to the following updates and fixes for December 2023:

## Bug fixes

> The following are updates for the dbt Semantic Layer and MetricFlow:

Author: ```suggestion
The following are updates for the dbt Semantic Layer:
```

---
title: "Updates and fixes: dbt Semantic Layer and MetricFlow updates for December 2023."
description: "December 2023: Enhanced Tableau integration, BIGINT support, LookML to MetricFlow conversion, and deprecation of legacy features."
sidebar_label: "Update ad fixes: dbt Semantic Layer and MetricFlow."
sidebar_position: 08
date: 2023-12-22
---
The dbt Labs team continues to work on adding new features, fixing bugs, and increasing reliability for the dbt Semantic Layer and MetricFlow. Here are the updates and fixes for December 2023.

## Bug fixes

> - The dbt Semantic Layer integration with Tableau now supports queries that resolve to a "NOT IN" clause (for example: using "exclude" in the filtering user interface). Previously it wasn’t supported.

Author: do we need to give more info on the memory leak?

Reviewer: I don't think so.... the errors that customers might have experienced from this could be different. We could maybe include a time period of when people might have been affected, but this could be hard to find.

## Improvements

- Deprecation &mdash; We deprecated [dbt Metrics and the legacy dbt Semantic Layer](/docs/dbt-versions/release-notes/Dec-2023/legacy-sl), both supported on dbt version 1.5 or lower. This change came into effect on December 15th, 2023.
- Improved dbt converter tool &mdash; The [dbt converter tool](https://github.com/dbt-labs/dbt-converter) can now help automate some of the work in converting from LookML (Looker's modeling language) for those who are migrating. Previously this wasn’t available. 

## New features

- Simplified group-by-item requests. We updated the way the MetricFlow query resolver finds queryable dimensions for metrics. The main improvements ares:
  - If the grain of a time dimension in a query is not specified, then the grain of the requested time dimension is resolved to be the finest grain that is available for the queried metrics. For example, say you have two metrics; revenue which has a weekly grain and orders which has a daily grain. If you query these metrics like this: `dbt sl query --metrics revenue,orders --group-by metric_time` metricflow will automatically query these metrics at a weekly grain.  
  

> - In a metric filter, if an ambiguous time dimension does not specify the grain, and all semantic models that are used to compute the metric define the time dimension with the same grain, MetricFlow should assume the specific time dimension is that grain. For example, say I have two metrics; revenue and users which are both daily. I can query these metrics without sepcifying the time dimension grain in the filte i.e `mf query --metrics users,revenue --group-by metric_time --where "{{ TimeDimension('metric_time') }} = '2017-07-30' "`

Author: ```suggestion
- Assumes time dimension grain: When using a metric filter, if an ambiguous time dimension doesn't specify the grain, and all used semantic models define this time dimension with the same grain, MetricFlow now automatically assumes the time dimension to be of that grain.
  - For example, if you have two daily metrics: `revenue` and `users` &mdash; you can now query these metrics without specifying the time dimension grain in the filter:  `mf query --metrics users,revenue --group-by metric_time --where "{{ TimeDimension('metric_time') }} = '2017-07-30' "`

```

---
title: "dbt Semantic Layer and MetricFlow updates for December 2023"
description: "December 2023: Enhanced Tableau integration, BIGINT support, LookML to MetricFlow conversion, and deprecation of legacy features."
sidebar_label: "Update and fixes: dbt Semantic Layer and MetricFlow"
sidebar_position: 08
date: 2023-12-22
---
The dbt Labs team continues to work on adding new features, fixing bugs, and increasing reliability for the dbt Semantic Layer and MetricFlow.

> Refer to the following updates and fixes for December 2023:

Author: ```suggestion
Refer to the following updates and fixes for December 2023.
```
